web/src/hotkey.js

```diff
@@ -386,6 +387,11 @@ function handle_popover_events(event_name) {
         return true;
     }
 
+    if (user_group_popover.is_open()) {
+        user_group_popover.handle_keyboard(event_name);
+        return true;
+    }
```

Comment:
I don't think this belongs in the first commit -- the first commit should be a pure code move with no functional changes, and if I read this correctly, this part is actually adding a keyboard navigation feature to this popover that previously didn't have one.

Reply:
This is the tricky part with this popover. The popover includes a keyboard navigation feature, but it was previously using keyboard handling functions from other popovers, making the code in `popovers.js` quite messy.

The tricky part is that I created `is_open` and `handle_keyboard`, but without them, this extraction commit would break those functionalities.

What are your thoughts on the best strategy to proceed?

Reply:
I went ahead and merged this extraction commit after revising the commit message to explain this detail. In the future, this sort of information is what the body of commit messages are for -- an explanation for why you made choices that are not consistent with what one would expect from reading the title.

That said, the best way to have done this series would in my view have been to do a first commit that fixes the variable/function reuse (with a nice commit message explaining it), and then a second commit that is a pure code move; that would have been significantly easier to review.

Reply:
Got it! Thanks!